### PR TITLE
fix(reporting): sanitize inputs on init

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

Sanitize inputs on initReport

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

jira details

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by a-launois — DISMISSED** [review]

**Review by AssanDiomande — DISMISSED** [review]

**Review by a-launois — DISMISSED** [review]

File: www/include/reporting/dashboard/initReport.php [hunk 1]
```diff
@@ -244,13 +244,33 @@
...
+$period_choice = (isset($_POST["period_choice"]))
+    ? filter_var($_POST["period_choice"], FILTER_SANITIZE_STRING)
+    : "preset";
+
+$period = '';
+if (isset($_POST['period'])) {
+    $period = filter_var($_POST['period'], FILTER_SANITIZE_STRING);
+} elseif (isset($_GET['period'])) {
+    $period = filter_var($_GET['period'], FILTER_SANITIZE_STRING);
+}
```
Comment (kduret) [inline www/include/reporting/dashboard/initReport.php:256]:
```suggestion
$period = filter_var(
    $_POST['period'] ?? $_GET['period'] ?? '',
    FILTER_SANITIZE_STRING
);
```

File: www/include/reporting/dashboard/initReport.php [hunk 2]
```diff
@@ -244,13 +244,33 @@
...
+
+$get_date_start = '';
+
+if (isset($_POST['StartDate'])) {
```
Comment (kduret) [inline www/include/reporting/dashboard/initReport.php:260]: same as above

File: www/include/reporting/dashboard/initReport.php [hunk 3]
```diff
@@ -244,13 +244,33 @@
...
+    $get_date_start = filter_var($_POST['StartDate'], FILTER_SANITIZE_STRING);
+} elseif (isset($_GET['StartDate'])) {
+    $get_date_start = filter_var($_GET['StartDate'], FILTER_SANITIZE_STRING);
+}
+
+$get_date_end = '';
+
+if (isset($_POST['EndDate'])) {
```
Comment (kduret) [inline www/include/reporting/dashboard/initReport.php:268]: same as above

File: www/include/reporting/dashboard/initReport.php [hunk 4]
```diff
@@ -244,13 +244,33 @@
...
-$period_choice = (isset($_POST["period_choice"])) ? $_POST["period_choice"] : "preset";
-$period = (isset($_POST["period"])) ? $_POST["period"] : "";
-$period = (isset($_GET["period"])) ? $_GET["period"] : $period;
-$get_date_start = (isset($_POST["StartDate"])) ? $_POST["StartDate"] : "";
-$get_date_start = (isset($_GET["start"])) ? $_GET["start"] : $get_date_start;
-$get_date_end = (isset($_POST["EndDate"])) ? $_POST["EndDate"] : "";
-$get_date_end = (isset($_GET["end"])) ? $_GET["end"] : $get_date_end;
+$period_choice = (isset($_POST["period_choice"]))
+    ? filter_var($_POST["period_choice"], FILTER_SANITIZE_STRING)
+    : "preset";
```
Comment (kduret) [inline www/include/reporting/dashboard/initReport.php:249]:
```suggestion
$period_choice = filter_var(
    $_POST["period_choice"] ?? 'preset',
    FILTER_SANITIZE_STRING
);
```

**Review by AssanDiomande — DISMISSED** [review]

**Comment by adr-mo** [comment]
@kduret all your feedbacks were applied 👍  thanks 💯

File: www/include/reporting/dashboard/initReport.php [hunk 5]
```diff
@@ -244,13 +244,32 @@
...
+    $_POST["period_choice"] ?? "preset",
+    FILTER_SANITIZE_STRING
+);
+
+$period = '';
+
+$period = filter_var(
+    $_POST['period'] ?? $_GET['period'],
+    FILTER_SANITIZE_STRING
+);
```
Comment (kduret) [inline www/include/reporting/dashboard/initReport.php:257]:
```suggestion
$period = filter_var(
    $_POST['period'] ?? $_GET['period'] ?? '',
    FILTER_SANITIZE_STRING
);
```

File: www/include/reporting/dashboard/initReport.php [hunk 6]
```diff
@@ -244,13 +244,32 @@
...
+
+$get_date_start = '';
+
+$get_date_start = filter_var(
+    $_POST['StartDate'] ?? $_GET['StartDate'],
+    FILTER_SANITIZE_STRING
+);
```
Comment (kduret) [inline www/include/reporting/dashboard/initReport.php:264]:
```suggestion
$get_date_start = filter_var(
    $_POST['StartDate'] ?? $_GET['StartDate'] ?? '',
    FILTER_SANITIZE_STRING
);
```

File: www/include/reporting/dashboard/initReport.php [hunk 7]
```diff
@@ -244,13 +244,32 @@
...
+
+$get_date_end = '';
+
+$get_date_start = filter_var(
+    $_POST['EndDate'] ?? $_GET['EndDate'],
+    FILTER_SANITIZE_STRING
+);
```
Comment (kduret) [inline www/include/reporting/dashboard/initReport.php:271]:
```suggestion
$get_date_start = filter_var(
    $_POST['EndDate'] ?? $_GET['EndDate'] ?? '',
    FILTER_SANITIZE_STRING
);
```

File: www/include/reporting/dashboard/initReport.php (hunk 6 above)
Comment (callapa) [inline www/include/reporting/dashboard/initReport.php:264]:
```suggestion
$get_date_start = filter_var(
    $_GET['start'] ?? $_POST['StartDate'] ?? '',
    FILTER_SANITIZE_STRING
);
```

**Review by callapa — CHANGES_REQUESTED** [review]

File: www/include/reporting/dashboard/initReport.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (adr-mo) [reply www/include/reporting/dashboard/initReport.php]: Done ✅  thanks (there was also a bad copy/paste on the variable name)

**Review by kduret — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

**Review by kduret — APPROVED** [review]

**Review by AssanDiomande — APPROVED** [review]

**Pull request merged**